src/etl/transform/standard_model/concept_schema.py

```diff
@@ -222,41 +223,96 @@ class C:
     CONCEPT.DIAGNOSIS,
     CONCEPT.OUTCOME,
     CONCEPT.GENOMIC_FILE,
-    CONCEPT.READ_GROUP
+    CONCEPT.READ_GROUP,
+    CONCEPT.BIOSPECIMEN_GENOMIC_FILE,
+    CONCEPT.BIOSPECIMEN_DIAGNOSIS,
+    CONCEPT.READ_GROUP_GENOMIC_FILE,
 }
 
 
-def _create_identifiers():
+def _create_unique_key_composition():
     """
-    Build the set of identifying concept properties.
-
-    - All CONCEPT.<>.ID properties are identifiers
-    - Some concepts have additional identifying properties
+    Build a dict which stores the attributes used to build a unique key
+    for a particular concept. This key uniquely identifies concept instances of
+    the same type.
+
+    A key in the dict is a standard concept and a value in
+    the dict is a set of concept attributes.
+
+    - The ID attribute is usually populated with a unique identifier
+    provided from the source data and so the default unique key for a concept
+    is composed of just the ID attribute.
+
+    - However, some concepts don't typically have a unique ID assigned in
+    the source data. These concepts can be uniquely identified by a
+    combination of one or more other concept attributes. Thus, the unique key
+    would be composed of a set of attribute values joined together by some
+    delimiter.
+
+    - For example a phenotype doesn't typically have a unique identifier
+    assigned to it in the source data. But a phenotype can be uniquely
+    identified by a combination of the participant's id, phenotype name,
+    observed status, and the age of the participant when the observation was
+    recorded. The values for these attributes would form the unique key for a
+    phenotype instance.
     """
+    # Default unique keys
     identifiers = {}
     for concept in concept_set:
-        identifiers[concept] = {concept.ID}
-
-    # Add other misc identifiers
-    identifiers[CONCEPT.GENOMIC_FILE].add(CONCEPT.GENOMIC_FILE.FILE_PATH)
-    identifiers[CONCEPT.SEQUENCING].add(CONCEPT.SEQUENCING.LIBRARY_NAME)
-    identifiers[CONCEPT.BIOSPECIMEN].add(CONCEPT.BIOSPECIMEN.ALIQUOT_ID)
+        identifiers[concept._CONCEPT_NAME] = [concept.ID]
+
+    # Compound unique keys
```

Comment:
I think we still need the `# Add other misc identifiers` section that got removed, which means that the new construction that converts sets to lists may not work. It probably needs to be a set of lists.